File: spec/python/test_recursive_one.py
import unittest

from recursive_one import RecursiveOne

class TestRecursiveOne(unittest.TestCase):
    def test_recursive_one(self):
        r = RecursiveOne.from_file("src/fixed_struct.bin")

        self.assertEqual(r.one, 0x50)
        self.assertEqual(r.next.one, 0x41)
        self.assertEqual(r.next.next.one, 0x43)
        self.assertEqual(r.next.next.next.finisher, 0x2d4b)

File: spec/lua/test_recursive_one.lua
local luaunit = require("luaunit")

require("recursive_one")

TestRecursiveOne = {}

function TestRecursiveOne:test_recursive_one()
    local r = RecursiveOne:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.one, 0x50)
    luaunit.assertEquals(r.next.one, 0x41)
    luaunit.assertEquals(r.next.next.one, 0x43)
    luaunit.assertEquals(r.next.next.next.finisher, 0x2d4b)
end
